pkg/api/customization/namespace/link.go

```diff
@@ -0,0 +1,139 @@
+package namespace
+
+import (
+	"github.com/rancher/norman/httperror"
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/pkg/clustermanager"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/kubernetes/pkg/printers"
+	"strings"
+)
+
+var ExportResourcePrefixMappings = map[string]string{
+	"pods":                   "api/v1",
+	"configmaps":             "api/v1",
+	"services":               "api/v1",
+	"replicationcontrollers": "api/v1",
+	"deployments":            "apis/extensions/v1beta1",
+	"daemonsets":             "apis/extensions/v1beta1",
+	"replicasets":            "apis/extensions/v1beta1",
+	"statefulsets":           "apis/apps/v1beta1",
+	"jobs":                   "apis/batch/v1",
+	"cronjobs":               "apis/batch/v1beta1",
+}
+
+var ExportPrinters = map[string]printers.ResourcePrinter{
+	"json": &printers.JSONPrinter{},
+	"yaml": &printers.YAMLPrinter{},
+}
+
+func NewLinkHandler(next types.RequestHandler, manager *clustermanager.Manager, management *config.ScaledContext) types.RequestHandler {
+
+	lh := &yamlLinkHandler{
+		next:              next,
+		clusterManagement: manager,
+		clusterLister:     management.Management.Clusters("").Controller().Lister(),
+	}
+
+	return lh.LinkHandler
+}
+
+type yamlLinkHandler struct {
+	next              types.RequestHandler
+	clusterManagement *clustermanager.Manager
+	clusterLister     v3.ClusterLister
+}
+
+func (s *yamlLinkHandler) callNext(apiContext *types.APIContext, next types.RequestHandler) error {
+	if s.next != nil {
+		return s.next(apiContext, next)
+	} else if next != nil {
+		return next(apiContext, nil)
+	}
+
+	return httperror.NewAPIError(httperror.NotFound, "link not found")
+}
+
+func (s *yamlLinkHandler) LinkHandler(apiContext *types.APIContext, next types.RequestHandler) error {
+	if apiContext.Link != "yaml" {
+		return s.callNext(apiContext, next)
+	}
+
+	clusterName := s.clusterManagement.ClusterName(apiContext)
+	cluster, err := s.clusterLister.Get("", clusterName)
+	if err != nil {
+		return err
+	}
+	userContext, err := s.clusterManagement.UserContext(clusterName)
+	if err != nil {
+		return err
+	}
+
+	ns := apiContext.ID
+	result := &unstructured.UnstructuredList{}
+	result.SetAPIVersion("v1")
+	result.SetKind("List")
+
+	resources := apiContext.Request.URL.Query()["resource"]
+	toExportResourceMappings := getResourcePrefixMap(resources)
+
+	for kind, prefix := range toExportResourceMappings {
+
+		req := userContext.UnversionedClient.Get().Prefix(prefix).Namespace(ns).Resource(kind)
+		for k, v := range apiContext.Request.URL.Query() {
+			req.Param(k, strings.Join(v, ","))
+		}
+		for k, v := range apiContext.Request.Header {
+			req.SetHeader(k, v...)
+		}
+		req.SetHeader("Accept", "*/*")
+		req.SetHeader("Authorization", "Bearer "+cluster.Status.ServiceAccountToken)
```

Comment:
Please don't copy the ServiceAccountToken, just skip copying Authorization header from the user.

Reply:
ok